extensions/panache/hibernate-orm-panache-kotlin/runtime/src/main/kotlin/io/quarkus/hibernate/orm/panache/kotlin/PanacheQuery.kt

```diff
@@ -27,6 +28,13 @@ interface PanacheQuery<Entity: Any> {
      */
     fun <NewEntity: Any> project(type: Class<NewEntity>): PanacheQuery<NewEntity>
 
+    /**
+     * Generates a SELECT clause with the specified properties. The row mapper function is used to build a DTO object for each row of the result.
+     *
+     * @return a new query with the same state as the previous one (params, page, range, lockMode, hints, ...).
+     */
+    fun <NewEntity: Any> project(rowMapper: Function<Array<Object>, NewEntity>, vararg properties: String): PanacheQuery<NewEntity>
```

Comment:
For the kotlin side, i would reverse the argument order.  kotlin doesn't require that `vararg` arguments go last and reversing the order would allow for more idiomatic kotlin like this:

```kotlin
val query = Person.find("order by height desc") 
                .project("name", "surname") { result -> 
                    User user = new User();
                    user.name = result[0] as String;
                    user.surname = result[1] as String;
                    return user;
                }
```

Reply:
Hm, I can change that but I thought that we should keep the kotlin API as close as possible.

Reply:
I try to do the same but sometimes it's "better" to make things idiomatic, imo.

Reply:
The main purpose of having a different interface for Kotlin and Java is to allow idiomatic Kotlin API so I'm +1 on this one.